packages/sysweb3-keyring/src/transactions/syscoin.ts

```diff
@@ -795,13 +835,151 @@ export class SyscoinTransactions implements ISyscoinTransactions {
     }
   };
 
+  public async sendRawTransaction(txHex: string, xpub: string, psbt: any) {
+    const { main } = this.getSigner();
+    const axiosConfig = {
+      headers: {
+        'X-User-Agent':
+          'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3',
+      },
+      withCredentials: true,
+    };
+    try {
+      let blockbookURL = main.blockbookURL.slice();
+      if (blockbookURL) {
+        blockbookURL = blockbookURL.replace(/\/$/, '');
+      }
+      // eslint-disable-next-line no-undef
+      if (fetch) {
+        const requestOptions = {
+          method: 'POST',
+          body: txHex,
+        };
+        // eslint-disable-next-line no-undef
+        const response = await fetch(
+          blockbookURL + '/api/v2/sendtx/',
+          requestOptions
+        );
+
+        if (response.ok) {
+          const data = await response.json();
+
+          await sys.utils.fetchBackendAccount(
+            blockbookURL,
+            xpub,
+            'tokens=used&details=tokens',
+            true
+          );
+
+          return data;
+        }
+      } else {
+        const request = await axios.post(
+          blockbookURL + '/api/v2/sendtx/',
+          txHex,
+          axiosConfig
+        );
+        if (request && request.data) {
+          await sys.utils.fetchBackendAccount(
+            blockbookURL,
+            xpub,
+            'tokens=used&details=tokens',
+            true
+          );
+          console.log(
+            'tx send successfully',
+            psbt.extractTransaction().getId()
+          );
+          return psbt;
+        }
+      }
+      return null;
+    } catch (e) {
+      return e;
+    }
+  }
+
   public confirmNativeTokenSend = async (
-    temporaryTransaction: ITokenSend
+    temporaryTransaction: ITokenSend,
+    isTrezor?: boolean
   ): Promise<ITxid> => {
     const { hd, main } = this.getSigner();
 
     const { receivingAddress, amount, fee } = temporaryTransaction;
 
+    if (isTrezor) {
+      try {
+        const { activeAccountId, accounts, activeAccountType, activeNetwork } =
+          this.getState();
+        const coin =
+          activeNetwork.currency && activeNetwork.currency.toLocaleLowerCase();
+        const {
+          xpub,
+          balances: { syscoin },
+        } = accounts[activeAccountType][activeAccountId];
+
+        const feeRate = new sys.utils.BN(fee * 1e8);
+
+        const value = new sys.utils.BN(amount * 1e8);
+
+        let outputs = [
+          {
+            address: receivingAddress,
+            value,
+          },
+        ];
+
+        const changeAddress = await this.trezor.getAddress({
```

Comment:
This should be fetched straight from the blockbook, we must only rely on trezor for fetching initially the account and signing the transaction